src/index.js

```diff
@@ -2,11 +2,25 @@
  * Validate an object is an instance of the expected type, or is a valid constructor object for the requested type
  * @param {*} value - The value for validation
  * @param {*} Type - The type for instance of comparison
- * @param {string} message - The message to be displayed if coercion fails
+ * @param {string} messageOrOptions - The message to be displayed if coercion fails
  * @returns {*} - Instance of provided Type
  * @throws {TypeError} - If the value is not coercable.
  */
-export function coerce (value, Type, message) {
+/**
+ * Validate an object is an instance of the expected type, or is a valid constructor object for the requested type
+ * @param {*} value - The value for validation
+ * @param {*} Type - The type for instance of comparison
+ * @param {string|object} messageOrOptions - Options for coerce. See below
+ * @param {object} messageOrOptions.parameterName - The name of the parameter being coerced
+ * @returns {*} - Instance of provided Type
+ * @throws {TypeError} - If the value is not coercable.
+ */
+export function coerce(value, Type, messageOrOptions) {
+    let errorMessage = messageOrOptions;
+    if (typeof messageOrOptions !== 'string') {
+        errorMessage = `${messageOrOptions.parameterName} should be coerceable to ${Type.name}. Provided value: ${JSON.stringify(value, null, 2)}`;
```

Comment:
I think I'd rather flip this so that the default value is the parameter name and it can be overridden with an options object.

Despite technically maintaining the same signature, it would constitute a breaking change -- but I think for the better.

Reply:
I'm not sure I completely follow. Can you show me example of the two method signatures your thinking of?

Reply:
I tend to prefer "ripping off the band-aid" to deprecating an API, but it's a common approach so it seemed responsible to suggest it 😉. 

Considering parameter validation on the options object will direct the consumer's migration effort, I'm comfortable with just taking the leap.

That said, I think it would behoove us to survey some of our colleagues for their views to make sure we're not out-of-step.